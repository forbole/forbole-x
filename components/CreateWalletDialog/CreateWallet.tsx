import {
  Button,
  DialogActions,
  DialogContent,
  DialogContentText,
  Typography,
  Box,
} from '@material-ui/core'
import useTranslation from 'next-translate/useTranslation'
import React from 'react'
import MnemonicPhraseInput from '../MnemonicPhraseInput'
import useStyles from './styles'

interface CreateWalletProps {
  mnemonic: string
  onConfirm(): void
}

const CreateWallet: React.FC<CreateWalletProps> = ({ mnemonic, onConfirm }) => {
  const { t } = useTranslation('common')
  const classes = useStyles()

  return (
    <>
      <DialogContent className={classes.dialogContent}>
        <DialogContentText>{t('create new wallet description')}</DialogContentText>
        <MnemonicPhraseInput disabled mnemonic={mnemonic} />
        <Box mt={6} />
      </DialogContent>
      <DialogActions>
        <Box flex={1} display="flex" flexDirection="column" mb={3}>
          <Button
            className={classes.button}
            variant="contained"
            color="primary"
            onClick={onConfirm}
          >
            {t('create new wallet button')}
          </Button>
          <Typography align="center">{t('create new wallet caption')}</Typography>
        </Box>
      </DialogActions>
    </>
  )
}

export default CreateWallet
