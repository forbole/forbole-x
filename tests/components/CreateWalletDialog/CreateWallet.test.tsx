import React from 'react';
import renderer from 'react-test-renderer';
import CreateWallet from '../../../components/CreateWalletDialog/CreateWallet';

const onConfirm = jest.fn();

jest.mock('../../../components/MnemonicPhraseInput', () => 'MnemonicPhraseInput');

describe('component: CreateWalletDialog - CreateWallet', () => {
  it('renders correctly', () => {
    const component = renderer.create(<CreateWallet onConfirm={onConfirm} mnemonic="mnemonic" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('calls onConfirm when button is clicked', () => {
    const component = renderer.create(<CreateWallet onConfirm={onConfirm} mnemonic="mnemonic" />);
    renderer.act(() => {
      component.root.findByType('form').props.onSubmit({ preventDefault: jest.fn() });
    });
    expect(onConfirm).toBeCalled();
  });
});

afterEach(() => {
  jest.clearAllMocks();
});
