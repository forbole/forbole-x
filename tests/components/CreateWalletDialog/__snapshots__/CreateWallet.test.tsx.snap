// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: CreateWalletDialog - CreateWallet renders correctly 1`] = `
Array [
  <div
    className="MuiDialogContent-root HookGlobalStyles-dialogContent-7"
  >
    <p
      className="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiTypography-colorTextSecondary"
    >
      common:create new wallet description
    </p>
    <MnemonicPhraseInput
      disabled={true}
      mnemonic="mnemonic"
    />
    <div
      className="MuiBox-root MuiBox-root-8"
    />
  </div>,
  <div
    className="MuiDialogActions-root MuiDialogActions-spacing"
  >
    <div
      className="MuiBox-root MuiBox-root-9"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained HookGlobalStyles-button-1 MuiButton-containedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[MockFunction]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          common:create new wallet button
        </span>
      </button>
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter"
      >
        common:create new wallet caption
      </p>
    </div>
  </div>,
]
`;
